### PR TITLE
Fix for two problems with format separators

### PR DESCRIPTION
**Pull request by burner**
fix issue 17459
fix issue 18047

This is in one commit as the two problem are located at the place

### CODE REVIEW

File: std/format.d [hunk 1]
```diff
@@ -1995,6 +1995,7 @@ private void formatUnsigned(Writer, T, Char)
             + zerofill
             + digits.length
             + ((fs.flSeparator != 0) * (digits.length / fs.separators))
+            - (fs.flSeparator != 0 && digits.length % fs.separators == 0)
```
Comment (andralex) [inline std/format.d]: `((fs.flSeparator != 0) * ((digits.length - 1) / fs.separators))` replaces the two lines and is quite a bit simpler

File: std/format.d [hunk 2]
```diff
@@ -2013,8 +2014,34 @@ private void formatUnsigned(Writer, T, Char)
...
+            if (j != fs.width && j % fs.separators == 0)
+            {
+                put(w, fs.separatorChar);
+                ++i;
+            }
+            if (i < zerofill)
+            {
+                   put(w, '0');
+            }
+            --j;
```
Comment (andralex) [inline std/format.d]: move to loop increment

File: std/format.d (hunk 1 above)
Comment (burner) [reply std/format.d]: fxied

File: std/format.d (hunk 2 above)
Comment (burner) [reply std/format.d]: and fixed

File: std/format.d [hunk 3]
```diff
@@ -2013,8 +2013,33 @@ private void formatUnsigned(Writer, T, Char)
...
-        put(w, '0');
+    if (fs.flSeparator)
+    {
+        if (zerofill > 0)
+        {
+            put(w, '0');
+            --zerofill;
+        }
+
+        int j = fs.width;
```
Comment (andralex) [inline std/format.d]: should this be uint?

Comment (burner) [reply std/format.d]: FormatSpec.width is an int. So I think int is good

**Pull request merged**